### DISCUSSION TITLE
Name Unassigned Error

### DISCUSSION THREAD

**Discussion by artsan43** [bengtmartensson/IrScrutinizer#436] [Export issues] [answered]
In the past I have successfully used this excellent program to generate a ccf containing all possible IR codes for a given protocol and device. After a long hiatus I have recently tried to do this again, but when I press "Export Generated" I get an error message; "NameUnassignedException: Name E not assigned". I have tried uninstalling and re-installing the program, to no avail. I must be doing something stupid, but I am at a loss. Can anyone shed any light on this error and what is causing it?

**Comment by bengtmartensson** [comment]
You are trying to render a protocol containing a named parameter "E" (Samsung36 is one such) without assigning a value to E. (So "Name E ot assigned" is accurate.) So you must assign an E value.

**Reply by artsan43** [reply]
Sorry, but I don't understand. What is an E value, and where would I input it? I am trying to generate a complete ccf for NEC2 protocol with D: 234 and S: 194. I put a * in the "F" field and left the "T" field blank. What am I doing wrong?

**Comment by bengtmartensson** [comment]
Make a screen shot.

**Comment by artsan43** [comment]
![image](https://user-images.githubusercontent.com/88629532/128645082-273f5200-b15f-421c-9f35-374b08bc4157.png)
or
[Doc1.docx](https://github.com/bengtmartensson/IrScrutinizer/files/6951423/Doc1.docx)

Forgive me...I'm not very sophisticated when it comes to this.

**Comment by bengtmartensson** [comment, marked as answer]
The selected protocol 48-NEC2 (which is a protocol very different from NEC2) _has_ an E parameter, so you have to assign it. Just type for example 

    E=42

into the "Additional paramerers" field. (Note that there are no spaces around the  equal sign.) (A parameter is "Additional" if it is not D, S, F or T.) Or select the real NEC2...

**Reply by artsan43** [reply]
Wow! That worked! No idea if the ccf contains the signals I need (for a Roku player) but I'll try it out. Thanks so much for your help. Your program is amazing.
